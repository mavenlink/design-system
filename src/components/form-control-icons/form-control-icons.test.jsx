import React from 'react';
import {
  render,
  screen,
} from '@testing-library/react';
import calendarSvg from '../../svgs/calendar.svg';
import Icon from '../icon/icon.jsx';
import FormControlIcons from './form-control-icons.jsx';

describe('<FormControlIcons>', () => {
  it('has defaults', () => {
    render(<FormControlIcons />);
    expect(document.body).toMatchSnapshot();
  });

  it('renders children', () => {
    render(<FormControlIcons><Icon icon={calendarSvg} label="kerfuffle" /></FormControlIcons>);
    expect(screen.getByTitle('kerfuffle')).toBeInTheDocument();
  });

  it('sets className', () => {
    const { container } = render(<FormControlIcons className="test-class" />);
    expect(container.querySelector('.test-class')).toBeInTheDocument();
  });

  it('shows caution icon if we have a validationMessage', () => {
    render(<FormControlIcons validationMessage="This is an error" />);
    expect(screen.getByTitle('This is an error')).toBeInTheDocument();
  });
});
