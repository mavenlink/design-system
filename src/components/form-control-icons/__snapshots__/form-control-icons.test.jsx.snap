// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormControlIcons> has defaults 1`] = `
<body>
  <div>
    <div
      class="icons"
    />
  </div>
</body>
`;
